apps/group/contracts/Group.sol

```diff
@@ -45,26 +69,113 @@ contract Group is App, Initializable, IForwarder, EVMCallScriptRunner {
     */
     function removeMember(address _entity) auth(REMOVE_MEMBER_ROLE) external {
         require(isGroupMember(_entity));
-        members[_entity] = false;
+        isMember[_entity] = false;
+        for (uint i = 0; i < members.length; i++) {
+            if (members[i] == _entity) {
+                members[i] = members[members.length - 1];
+                delete members[members.length - 1];
+                members.length -= 1;
+                break;
+            }
+        }
+
+        if (required > members.length)
+            changeRequirement(members.length);
+
         RemoveMember(_entity);
     }
 
+    function numMembers() public constant returns (uint) {
+        return members.length;
+    }
+
+    /**
+    * @notice Allows a group member to confirm that a specific action can be executed
+    * @param _actionId Index of an action
+    */
+    function confirm(uint _actionId) public {
+        require(isGroupMember(msg.sender));
+        confirmations[_actionId][msg.sender] = true;
+        Confirmation(_actionId, msg.sender);
```

Comment:
If confirmation makes the action approved, execution should be triggered to minimize the number of required transactions.

Reply:
Added some logic at the end of the function to execute the action if it has enough confirmations.